broker/machinebroker/server/machine_networkinterface_attach.go

```diff
@@ -71,31 +80,49 @@ func (s *Server) createIronCoreNetworkInterface(
 	cfg *IronCoreNetworkInterfaceConfig,
 ) (ironcoreMachineNic *computev1alpha1.NetworkInterface, aggIronCoreNic *AggregateIronCoreNetworkInterface, retErr error) {
 	log.V(1).Info("Getting network for handle")
-	ironcoreNetwork, err := s.networks.GetNetwork(ctx, cfg.NetworkID)
+
+	networkLabelsString, networkLabelsPresent := cfg.Attributes[machinepoolletv1alpha1.NetworkLabelsAttributeKey]
+	networkLabels, err := maps.UnmarshalLabels(networkLabelsString, networkLabelsPresent)
+	if err != nil {
+		return nil, nil, err
+	}
+	ironcoreNetwork, err := s.networks.GetNetwork(ctx, cfg.NetworkID, networkLabels)
```

Comment:
What happens here if we have two `NIC`s attached? Don't we create a second network since the `Labels` are then different? I think it will cause some errors